### ISSUE TITLE
Have to login twice.

### ISSUE THREAD

**Issue by deanfourie1** [brenner-tobias/addon-cloudflared#602]
### The problem

So basically, every time I login via Cloudflare, after about 1 minute it will log me out again and I have to login again, after that its fine.

So each time I login remote using the CF hostname, I have to login twice. I have had this issue for multiple previous versions as welll as the latest, 5.1.3

I am also having an issue where the first 2FA code entered is always reject, when I enter the 2FA code again, the second time (same code) it is accepted.

### What version of Cloudflared has the issue?

5.1.3

### What was the last working version of Cloudflared?

unsure

### What type of installation are you running?

Home Assistant Supervised

### Steps to reproduce the issue

Hard to reproduce I guess, but I just login and it logs me out.
Everything works 100%, except for me having to login twice.
I am also having an issue where the first 2FA code entered is always reject, when I enter the 2FA code again, the second time (same code) it is accepted.

**Comment by brenner-tobias** [comment]
Can you elaborate on your Cloudflare One set-up? Which login method / Identity Provider are you using?

**Comment by deanfourie1** [comment]
Thanks for the reply.

I am just using the method with a domain name, and the tunnel name. It auto created the tunnel and everything else and it worked.

Authenticated via the web browser.

**Comment by brenner-tobias** [comment]
Ok, got it, I was under the impression you were referring to some 3rd party login provider.
Is this the same if you login locally using the HA IP or some local hostname?

**Comment by 1nfinity84** [comment]
Oh that used to happen to me too (always on latest), now when the freenom domains went off I reinstalled the plugin and it stopped. Could be coincidence, but it'd usually go like this - I'd go to my domain, login, do something on dashboard, then for example went to node red and bam - login screen. It'd not do that in my local network.

**Comment by deanfourie1** [comment]
Thanks guys,

I am not using a domain from Freenom, I am using my own domain, hosted at onlydomains along with web hosting etc.

Cheers

**Comment by 1nfinity84** [comment]
weirdly enough, I had it happen today as well. Seems it's related to an add-on accessing something (maybe core files?). I tried to upload some files to www folder via the upload feature, before I selected the files I got logged out again. It doesn't bother me so much, since I rarely access via standard browser. It doesn't affect companion app at all. 
@brenner-tobias Tobias, is there anything I can do to help you figure this out?
